### PR TITLE
Replay bootstrapping trace

### PR DESCRIPTION
**Pull request by icristescu**
Initial benchmarks for replaying the trace of index operations during bootstrap. The trace is small for now, I’m working on getting a bigger one.

It needs the dependency on `tezos-context-hash` but only on the Hash module, and it needs to be in a separate package (https://github.com/icristescu/tezos-context-hash/tree/hash_pkg); Otherwise, we end up with the following dune error:

```
Error: Conflict between the following libraries:
- "index" in _build/default/src
- "index" in /Users/icristes/.opam/4.11.0/lib/index
  -> required by library "irmin-pack" in
     /Users/icristes/.opam/4.11.0/lib/irmin-pack
  -> required by library "tezos-context-hash" in
     _build/default/vendors/tezos-context-hash/src
-> required by executable replay in bench/dune:25

```

### CODE REVIEW

File: .gitmodules
Comment (Ngoguey42) [inline .gitmodules:3]: I suppose this should be set to the right URL before merging

File: bench/replay.ml
Comment (Ngoguey42) [inline bench/replay.ml:199]: This is going to be rendered a lot. It might currently slow down the replay.

Try adding a `~sampling_interval` to `Progress_unix.counter`. Aim for 1 to 100 renders per sec.

File: bench/replay.ml
Comment (Ngoguey42) [inline bench/replay.ml:149]: Isn't the store `fresh`?

**Review by Ngoguey42 — COMMENTED** [review]
👍

File: bench/replay.ml
Comment (CraigFe) [reply bench/replay.ml:199]: (Aside: I will fix this tediousness in the next release of Progress...)

**Comment by samoht** [comment]
This is blocked on @CraigFe's reorganisation of tezos-context-hash into different libraries.

**Comment by CraigFe** [comment]
Thanks! I've pushed a suggested alternative to the Hash extraction [here](https://github.com/tarides/tezos-context-hash/pull/26), which is very similar to yours, and PRed the necessary changes to this patch [here](https://github.com/icristescu/index/pull/2).

**Comment by CraigFe** [comment]
I'm not sure why the CI is failing to install `digestif` for `tezos-context-hash`; will ask around.

File: index-bench.opam
Comment (kit-ty-kate) [inline index-bench.opam:25]:
```suggestion
  "rusage" {>= "1.0.0" & with-test}
  "digestif" {>= "0.9.0"}
```

**Review by CraigFe — APPROVED** [review]
Thanks all :-) The CI is now happy, so I propose we merge this.

**Pull request merged**
